src/Mixin.php

```diff
@@ -0,0 +1,800 @@
+ * @method static void nullOrLengthBetween($value, $min, $max, $message = '')
+ * @method static void allLengthBetween($value, $min, $max, $message = '')
+ * @method static void nullOrFileExists($value, $message = '')
+ * @method static void allFileExists($value, $message = '')
+ * @method static void nullOrFile($value, $message = '')
+ * @method static void allFile($value, $message = '')
+ * @method static void nullOrDirectory($value, $message = '')
+ * @method static void allDirectory($value, $message = '')
+ * @method static void nullOrReadable($value, $message = '')
+ * @method static void allReadable($value, $message = '')
+ * @method static void nullOrWritable($value, $message = '')
+ * @method static void allWritable($value, $message = '')
+ * @method static void nullOrInterfaceExists($value, $message = '')
+ * @method static void allInterfaceExists($value, $message = '')
+ * @method static void nullOrPropertyExists($classOrObject, $property, $message = '')
+ * @method static void allPropertyExists($classOrObject, $property, $message = '')
+ * @method static void nullOrPropertyNotExists($classOrObject, $property, $message = '')
+ * @method static void allPropertyNotExists($classOrObject, $property, $message = '')
+ * @method static void nullOrMethodExists($classOrObject, $method, $message = '')
+ * @method static void allMethodExists($classOrObject, $method, $message = '')
+ * @method static void nullOrMethodNotExists($classOrObject, $method, $message = '')
+ * @method static void allMethodNotExists($classOrObject, $method, $message = '')
+ * @method static void nullOrKeyExists($array, $key, $message = '')
+ * @method static void allKeyExists($array, $key, $message = '')
+ * @method static void nullOrKeyNotExists($array, $key, $message = '')
+ * @method static void allKeyNotExists($array, $key, $message = '')
+ * @method static void nullOrCount($array, $number, $message = '')
+ * @method static void allCount($array, $number, $message = '')
+ * @method static void nullOrMinCount($array, $min, $message = '')
+ * @method static void allMinCount($array, $min, $message = '')
+ * @method static void nullOrMaxCount($array, $max, $message = '')
+ * @method static void allMaxCount($array, $max, $message = '')
+ * @method static void nullOrCountBetween($array, $min, $max, $message = '')
+ * @method static void allCountBetween($array, $min, $max, $message = '')
+ * @method static void nullOrIsNonEmptyList($array, $message = '')
+ * @method static void allIsNonEmptyList($array, $message = '')
+ * @method static void nullOrIsNonEmptyMap($array, $message = '')
+ * @method static void allIsNonEmptyMap($array, $message = '')
+ * @method static void nullOrUuid($value, $message = '')
+ * @method static void allUuid($value, $message = '')
+ * @method static void nullOrThrows($expression, $class = 'Exception', $message = '')
+ * @method static void allThrows($expression, $class = 'Exception', $message = '')
+ */
+interface Mixin
```

Comment:
I'm a fan of the `Mixin` Interface, and looking at it the methods are right.

I do think we can now drop the `@method` annotations on top of it, as those are already part of the interface now.

Reply:
I think we should add the methods that dont' have psalm assertions here as well, to drop the annotations, and keep everything working the same.

Reply:
I agree.

@Ocramius thoughts?

Reply:
Seems reasonable :+1:

Reply:
Done.